csharp/src/Apache.Arrow/Arrays/ArrowArrayFactory.cs

```diff
@@ -104,14 +104,6 @@ public static IArrowArray BuildArray(ArrayData data)
 
         public static IArrowArray Slice(IArrowArray array, int offset, int length)
         {
-            if (offset > array.Length)
```

Comment:
This check and the `length = Math.Min(...` adjustment below are also both accounted for in `ArrayData.Slice`, so I removed them too.